### DISCUSSION TITLE
Feature - compare one across others

### DISCUSSION THREAD

**Discussion by igoririsproject** [dodona-edu/dolos#973] [Ideas]
Now when selecting multiple files for comparison we see comparison results of all possible pairs.

Is it possible to compare one file with all others. For example:
Take `file1.js` and compare agains `file2.js`, `file3.js`, `file4.js`. So there is no need to do comparisons of files 2-4 with each other.
 
I did not find any way how to do this right now, so it would be great if there would be such a possiblity

**Comment by rien** [comment]
Hi, thanks for your suggestion!

What is the situation where you need this? Do you want to compare one student's submission with those of others? Or do you want to compare a teacher solution with that of students?

Are you aware of the submission detail page (this page: https://dolos.ugent.be/demo/pyramidal-constants/exercise/#/submissions/355) where you have a list of all similar files? If so, what are you lacking on that page?

How would you imagine comparing multiple files would look like?

**Reply by igoririsproject** [reply]
Yes, I used submissions, pair view and it's great.
The thing I would like to see - is to create pairs only for one file agaist others.
For example, I already checked works of several students against each other. And I have another result (file), which I would like to check against previous sumbissions. I don't want to check all previous submissions again, just the new one.

**Reply by rien** [reply]
This is indeed something that would be useful, especially for doing an incremental analysis (already discussed in #732). We won't implement this feature in the near future, but we would like to add support for this kind of checks in the future!

We will focus on the performance of Dolos next, reducing the call stack issues and speeding up an analysis as well.

**Comment by igoririsproject** [comment]
This could also partly solve the problem with `maximum call stack size exceeded`, mentioned [here](https://github.com/dodona-edu/dolos/discussions/969)
